### ISSUE TITLE
Feature for localization support using optional toLocaleString

### ISSUE THREAD

**Issue by clytras** [75lb/byte-size#9]
Hello and thanks for the conversion utility.

I want to add a localization functionality (and make a PR if we come up with a solution) and the simplest way that I can think of and keep backward compatibility, is to add an options flag that will indicate the usage of `toLocaleString` if available, else it will fall back using the `toFixed` function.

Example options:

```js
options = Object.assign({
  units: 'metric',
  precision: 1,
  localizedString: false,
}, defaultOptions, options)
```

Example `localizedString` option usage:

```diff
if (units) {
-  const value = units.from === 0
-    ? prefix + bytes
-    : prefix + (bytes / units.from).toFixed(options.precision)
-  this.value = value
+  if (units.from === 0) {
+    this.value = prefix + bytes
+  } else if (options.localizedString && typeof Intl === 'object' && typeof Intl.NumberFormat === 'function') {
+    this.value = prefix + (bytes / units.from).toLocaleString(undefined, {
+      minimumFractionDigits: options.precision,
+      maximumFractionDigits: options.precision,
+    })
+  } else {
+    this.value = prefix + (bytes / units.from).toFixed(options.precision)
+  }
  ...
} else { ... }
```

Let me know if something like this would make a nice addition to this utility.

**Comment by 75lb** [comment]
Yes, this feature is required.. I will have a quick look into your solution and get back to you, thanks..

**Comment by clytras** [comment]
Thank you @75lb.

Maybe we should add one more option to let the developers specify the locales (*`string` or `array` or `undefined`*) if they need to explicit set something other than the default browser locale (`undefined`).

```js
options = Object.assign({
  units: 'metric',
  precision: 1,
  localizedString: false,
  // We don't have to explicit set an undefined property,
  // we do it here to indicate there is such an option
  locales: undefined,
}, defaultOptions, options)
```

According [`Number.prototype.toLocaleString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString#using_locales), it can be a `string`, an `array` or `undefined`.

And then we use it with `toLocaleString`:

```js
(bytes / units.from).toLocaleString(options.locales, { ... })
```

**Comment by 75lb** [comment]
Hi @clytras .. I pushed a few separate, unrelated changes to `master` that I'd like in the next release.. To the [next](https://github.com/75lb/byte-size/tree/next) branch, I added my "starter" implementation for this feature.

Please checkout the `next` branch and check it covers everything. If it doesn't, please submit a PR with your ideas (don't worry about submitting the ultimate perfect PR, I will help get it through if necessary). 

I think the lib should be localised by default, so we don't need the `localizedString` flag..

**Comment by clytras** [comment]
It works like a charm with the default locale and with a custom defined locale.

My only concern is regarding backward compatibility on environments that don't have `Intl.NumberFormat` support, there won't be a fallback using the `toFixed`, but I believe this will be a next major version release maybe, which will have additional information that the environment must have support for `Intl.NumberFormat`. But overall, this won't be a big issue since most browsers and NodeJS have nice support of `Intl.NumberFormat` for quite a while now.

Thank you so much for making this useful utility and fast responding to such a feature request 👍🍺🍻

**Comment by 75lb** [comment]
Released in [v8.0.0](https://github.com/75lb/byte-size/releases/tag/v8.0.0)
